File: atividade-6/atividade.go
package main

import "fmt"

type Motor struct {
	Potencia int
}

type Carro struct {
	Marca  string
	Modelo string
	Motor  Motor
}

func (c Carro) ExibirDetalhes() {
	fmt.Printf("Carro: %s %s, Potência: %d HP\n", c.Marca, c.Modelo, c.Motor.Potencia)
}

func main() {
	motor := Motor{Potencia: 150}
	carro := Carro{Marca: "Toyota", Modelo: "Corolla", Motor: motor}
	carro.ExibirDetalhes()
}

File: atividade-6/atividade.py
class Motor:
    def __init__(self, potencia):
        self.potencia = potencia

class Carro:
    def __init__(self, marca, modelo, motor):
        self.marca = marca
        self.modelo = modelo
        self.motor = motor

    def exibir_detalhes(self):
        print(f"Carro: {self.marca} {self.modelo}, Potência: {self.motor.potencia} HP")

motor = Motor(150)
carro = Carro("Toyota", "Corolla", motor)
carro.exibir_detalhes()